File: fronter/src/model/memory.ts

export interface Memory {
  id: number;
  user_id: number;
  source_type: string;
  title: string;
  author: string;
  notes: string;
  factor: string;
  process: string;
  evaluation_axis: string;
  information_amount: string;
  tags: string;
  read_status: string;
  read_date: string | null;
  created_at: string;
  updated_at: string;
}

export interface AddMemory {
  title: string;
  user_id: number;
  author: string;
  source_type: string;
  notes: string;
  factor: string;
  process: string;
  evaluation_axis: string;
  information_amount: string;
  tags: string;
  read_status: string;
  read_date: string;
}

File: backer/godotask/infrastructure/db/model/memory.go
package model

import (
	"time"
)

type Memory struct {
	ID         int       `gorm:"primaryKey" json:"id"`
	UserID     int       `json:"user_id"`
	SourceType string    `json:"source_type" gorm:"default:book"`
	Title      string    `json:"title"`
	Author     string    `json:"author"`
	Notes      string    `json:"notes"`
	Tags       string    `json:"tags"`
	ReadStatus string    `json:"read_status" gorm:"default:unread"`
	ReadDate   *time.Time `json:"read_date"`
	Factor           string    `json:"factor"`
	Process          string    `json:"process"`
	EvaluationAxis   string    `json:"evaluation_axis"`
	InformationAmount string   `json:"information_amount"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

func (Memory) TableName() string {
  return "memories"
}